File: tests/__tests__/binary_trees/utility.js
import { BinaryTree } from "@/binary_trees/assets/BinaryTree.js";

export const buildBT = (nodes, rootId) => {
  const idToNode = {};

  nodes.forEach((node) => {
    idToNode[node.id] = new BinaryTree(node.value);
  });

  nodes.forEach((node) => {
    const btNode = idToNode[node.id];
    btNode.left = node.left ? idToNode[node.left] : null;
    btNode.right = node.right ? idToNode[node.right] : null;
  });

  return idToNode[rootId];
};

File: tests/python/binary_trees/utility.py
from problems.binary_trees.assets.BinaryTree import BinaryTree


def build_bt(nodes: list[dict], root_id: str) -> BinaryTree:
    id_to_node = {}

    for node in nodes:
        id_to_node[node["id"]] = BinaryTree(node["value"])

    for node in nodes:
        bt_node = id_to_node[node["id"]]
        bt_node.left = id_to_node.get(node["left"])
        bt_node.right = id_to_node.get(node["right"])

    return id_to_node[root_id]
